include/data_sink.hpp

```diff
@@ -72,6 +76,14 @@ class data_sink : public node<data_sink<T>> {
     std::vector<listener> listeners;
 
 public:
+    // TODO sink should register itself on construction, but name is set afterwards via
+    // set_name, which we have no hook into. Maybe the registration should be done by the
+    // graph creating/destroying the sink instead?
```

Comment:
I added another PR https://github.com/fair-acc/graph-prototype/pull/84 that should provide each node with a unique name... maybe this helps?